stdlib/database/h2/src/test/resources/test-src/sql/sql_actions_negative_test.bal

```diff
@@ -158,15 +257,102 @@ function testInvalidArrayofQueryParameters() returns string {
         if (j is json) {
             returnData = io:sprintf("%s", j);
         } else {
-            returnData = j.reason();
+            error e = j;
+            returnData = e.reason();
+        }
+    } else {
+        error e = x;
+        returnData = <string> e.detail().message;
+    }
+    checkpanic testDB.stop();
+    return returnData;
+}
+
+function testErrorWithInvalidArrayofQueryParameters() returns string {
+    h2:Client testDB = new({
+            path: "./target/tempdb/",
+            name: "TEST_SQL_CONNECTOR_H2",
+            username: "SA",
+            password: "",
+            poolOptions: { maximumPoolSize: 1 }
+        });
+
+    string returnData = "";
+    xml x1 = xml `<book>The Lost World</book>`;
+    xml x2 = xml `<book>The Lost World2</book>`;
+    xml[] xmlDataArray = [x1, x2];
+    sql:Parameter para0 = { sqlType: sql:TYPE_INTEGER, value: xmlDataArray };
+    var x = testDB->select("SELECT FirstName from Customers where registrationID in (?)", (), para0);
+
+    if (x is table<record {}>) {
+        var j = json.convert(x);
+        if (j is json) {
+            returnData = io:sprintf("%s", j);
+        } else {
+            error e = j;
+            returnData = e.reason();
         }
     } else {
-        returnData = <string> x.detail().message;
+        error e = x;
+        returnData = io:sprintf("%s", e);
     }
     checkpanic testDB.stop();
     return returnData;
 }
 
+function testCheckApplicationErrorType() returns [boolean, boolean] {
+    h2:Client testDB = new({
+            path: "./target/tempdb/",
+            name: "TEST_SQL_CONNECTOR_H2",
+            username: "SA",
+            password: "",
+            poolOptions: { maximumPoolSize: 1 }
+        });
+
+    string returnData = "";
+    xml x1 = xml `<book>The Lost World</book>`;
+    xml x2 = xml `<book>The Lost World2</book>`;
+    xml[] xmlDataArray = [x1, x2];
+    sql:Parameter para0 = { sqlType: sql:TYPE_INTEGER, value: xmlDataArray };
+    var x = testDB->select("SELECT FirstName from Customers where registrationID in (?)", (), para0);
+
+    boolean isJdbcClientError = false;
+    boolean isApplicationError = false;
+
+    if (x is sql:JdbcClientError) {
+        isJdbcClientError = true;
+        if (x is sql:ApplicationError) {
```

Comment:
Instead of having this within the `if (x is sql:JdbcClientError)` block, shall we have this check outside? 
Also lets check this for `error` as well.